### PR TITLE
=per #19283 Make LevelDB imports optional and update version dependencies

### PR DESCRIPTION
**Pull request by briantopping**

### CODE REVIEW

**Comment by ktoso** [comment]
This one is to the wrong branch though :-)
The previous one: https://github.com/akka/akka/pull/19284 is OK :)

File: project/AkkaBuild.scala [hunk 1]
```diff
@@ -1480,9 +1480,10 @@ object AkkaBuild extends Build {
     )
     def defaultImports = Seq("!sun.misc", akkaImport(), configImport(), scalaImport(), "*")
     def akkaImport(packageName: String = "akka.*") = versionedImport(packageName, "2.3", "2.4")
-    def streamAndHttpImport(packageName: String) = versionedImport(packageName, "1.0", "1.1")
+    def streamAndHttpImport(packageName: String) = versionedImport(packageName, "2.0", "2.1")
```
Comment (briantopping) [inline project/AkkaBuild.scala]: @ktoso I couldn't find this change on master. It definitely needs to be there, will look again after some sleep, want to get it in at the same time. Thanks!

**Comment by briantopping** [comment]
@ktoso: I just had a chance to read the branches explanation. This patch definitely needs to be applied, http-core can't find streams under OSGi because the embedded version number is too restrictive.

File: project/AkkaBuild.scala (hunk 1 above)
Comment (ktoso) [inline project/AkkaBuild.scala]: Ok, sure. This ccommit on this branch however must not touch the persistence project (see above)

File: project/AkkaBuild.scala [hunk 2]
```diff
@@ -1456,7 +1456,7 @@ object AkkaBuild extends Build {
 
     val transactor = exports(Seq("akka.transactor.*"))
 
-    val persistence = exports(Seq("akka.persistence.*"), imports = Seq(protobufImport()))
+    val persistence = exports(Seq("akka.persistence.*"), imports = Seq(protobufImport()) ++ leveldbImport())
```
Comment (ktoso) [inline project/AkkaBuild.scala]: Can not touch persistence on this branch.

**Comment by briantopping** [comment]
Ok, removed the akka-persistence changes. Is the etiquette to squash only after others have reviewed the requested change or is it okay to do it now?

**Comment by ktoso** [comment]
OK TO TEST

File: project/AkkaBuild.scala (hunk 1 above)
Comment (ktoso) [inline project/AkkaBuild.scala]: Thanks for removing the persistence bit.
This looks good to me, thanks!

Please squash and I'll merge.

**Comment by ktoso** [comment]
Oh, one thing – please don't mention LevelDB in the commit message here, that was the other part (already on master), thanks @briantopping !

**Comment by briantopping** [comment]
@ktoso Thanks! Squashed and commit message changed.

**Comment by ktoso** [comment]
Thanks for squashing @briantopping - would have been be nice to have kept the commit message to include the `=per #19283 ...` part, that's helpful when grepping logs, could you amend that and I merge right away?

**Comment by ktoso** [comment]
Refs #19283

**Comment by ktoso** [comment]
Merged manually with fixed title: https://github.com/akka/akka/commit/7cf8527bcb055bb12176601811b01de6f2197cdd

**Comment by briantopping** [comment]
Hey thanks Konrad! Please don't hesitate to pass along my willingness to help to the rest of the team, will do my best. Apologies I wasn't online when you were, I'm in GMT-7 right now, kind of tangential to working hours in EU.... cheers!

**Comment by ktoso** [comment]
Will do, thanks a lot!

-- 
Cheers,
Konrad 'ktoso’ Malawski
Akka @ Typesafe

On 30 December 2015 at 06:35:54, Brian Topping (notifications@github.com) wrote:

Hey thanks Konrad! Please don't hesitate to pass along my willingness to help to the rest of the team, will do my best. Apologies I wasn't online when you were, I'm in GMT-7 right now, kind of tangential to working hours in EU.... cheers!

**Pull request closed without merging**